### DISCUSSION TITLE
Can you download the app page(s) as a pdf file?

### DISCUSSION THREAD

**Discussion by sugatoray** [h2oai/wave#1656] [Q&A]
It will be nice to have a functionality on the app page to download the dashboard as a pdf file (preferably) or image file (png/svg: at least). I could not find it in the docs. Is there any such functionality already available in h2o-wave?

Thank you

**Comment by mturoci** [comment]
Hi @sugatoray, unfortunately, this is currently not possible, but tracked in https://github.com/h2oai/wave/issues/486.

Edit: Hitting cmd + P (and I assume ctr + P on windows/linux) should give you this feature for free.


https://user-images.githubusercontent.com/64769322/193999752-019ba952-66ac-4160-b2de-c3f86830bce3.mov

**Reply by sugatoray** [reply]
@mturoci Thank you for the quick update. The esthetic appeal of the dashboard gets completely lost if one tries to print to a pdf. 
 
As a possible stopgap: Could `playwright` be used to grab a screenshot of the dashboard and then save the screenshot as a PDF file, if someone clicks on a download button?

**Reply by mturoci** [reply]
> Could playwright be used to grab a screenshot of the dashboard and then save the screenshot as a PDF file, if someone clicks on a download button?

Definitely, but this is something that you need to do yourself as it's not part of the Wave itself.
